internal/backend/runtime/omni/controllers/omni/cluster_machine_status.go

```diff
@@ -188,8 +188,7 @@ func NewClusterMachineStatusController() *ClusterMachineStatusController {
 				case machineapi.MachineStatusEvent_RUNNING:
 					cmsVal.Stage = specs.ClusterMachineStatusSpec_RUNNING
 				case machineapi.MachineStatusEvent_MAINTENANCE:
-					// shouldn't happen as machine config should be submitted
-					cmsVal.Stage = specs.ClusterMachineStatusSpec_UNKNOWN
+					cmsVal.Stage = specs.ClusterMachineStatusSpec_CONFIGURING
```

Comment:
Now the machines won't be in Unknown state when they're added to the cluster and are still in maintenance.
I've noticed that Talos might have a slight delay before it reports `BOOTING` stage.